packages/cletus/src/agents/chat-agent.ts

```diff
@@ -262,25 +255,65 @@ Files:
 - If a file does not exist because you assumed the user gave a correct path, search for the file.
 - Before performing write operations, ensure you have the latest file state by reading it first.
 - Do not ask the user to locate a file unless you've already tried searching for it.
+
+Tools:
+- If you need tools from a different toolset, use the 'retool' utility to switch or enable adaptive selection.
+- The adaptive tool selection uses embedding similarity to find the most relevant tools for your current task.
+- Use 'retool' with null to re-enable adaptive selection if you've been focused on a specific toolset.
 </rules>
+
+<importantRules>
+- Don't present the results of an operation in <input> or <output> tags - those are only for your internal processing.
+- Don't ask for permission to perform operations - if you need to do something, just do it. The user will be asked for approval automatically if needed.
+- Todos are exclusively for Cletus's internal management of user requests. They are only referred to as todos - anything else should be assumed to be a separate data type.
+- If you've executed ANY tools - DO NOT ask a question at the end of your response. You are either going to automatically continue your work OR the user will respond next. NEVER ask a question after executing tools. Only for clarifications.
+</importantRules>
 `,
-    tools: [delegate, ...utilityTools],
+    // Dynamic tools based on adaptive selection
+    toolsFn: async (_, ctx) => {
```

Comment:
I don't think this is a function, look at prompt.ts file in core. You'll need to use retool